include/linux/soundwire/sdw_intel.h

```diff
@@ -4,6 +4,7 @@
 #ifndef __SDW_INTEL_H
 #define __SDW_INTEL_H
 
+#include <linux/acpi.h>
```

Comment:
interesting. @ujfalusi where do you include this sdw_intel.h file? I didn't see the link with this PR and I haven't seen any compilation errors?

Reply:
The last patch adds a new small driver: sound/soc/sof/intel/client-intel-probes.c
For the HDA probes support I need to include the hda.h, which will pulls in the sdw_intel.h.
I still needed to include the ../sof_priv.h in the driver because the hda.h would not compile due to missing definitions.

I'm not sure if it would be better if hda.h includes the ../sof_priv.h, in all C files it is is eventually got included either directly or indirectly via ../ops.h or other headers from the sof root.

Since we arrange headers in alphabetic orders, the sdw_intel.h tends to be quite late in the order and earlier includes will get the linux/acpi.h to not cause any issue.

Reply:
@plbossart, the linux/acpi.h has nothing to do with the driver in question. It should not include unrelevant include files because the given header would not compile otherwise.
The inclusion of ../sof-priv.h also have the same problem, it is there to make the hda.h to compile.

What could be done without fixing the header files:
A) create a new header file for this purpose only to hold the probes related definitions (hda_probes_ops only atm)
B)
```
#include <linux/auxiliary_bus.h>
#include <linux/module.h>
#include "../sof-client-probes.h"

/* ops for the probes support */
#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA_PROBES)
extern const struct sof_probes_ops hda_probe_ops;
#endif

static const struct auxiliary_device_id intel_probes_client_id_table[] = {
#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA_PROBES)
	{ .name = "snd_sof.hda-probes", .driver_data = (kernel_ulong_t)&hda_probe_ops, },
#endif
	{},
};
MODULE_DEVICE_TABLE(auxiliary, intel_probes_client_id_table);
```
I don't see how the hda.h can be split up in a meaningful way...